spring-integration-http/src/main/java/org/springframework/integration/http/outbound/AbstractHttpRequestExecutingMessageHandler.java

```diff
@@ -127,9 +125,25 @@ public AbstractHttpRequestExecutingMessageHandler(Expression uriExpression) {
 	 * <code>true</code>.
 	 * @param encodeUri true if the URI should be encoded.
 	 * @see UriComponentsBuilder
+	 * @deprecated since 5.3 in favor of {@link #setEncodingMode}
 	 */
+	@Deprecated
 	public void setEncodeUri(boolean encodeUri) {
-		this.encodeUri = encodeUri;
+		setEncodingMode(
+				encodeUri
+						? DefaultUriBuilderFactory.EncodingMode.TEMPLATE_AND_VALUES
+						: DefaultUriBuilderFactory.EncodingMode.NONE);
+	}
+
+	/**
+	 * Set the encoding mode to use.
+	 * By default this is set to {@link DefaultUriBuilderFactory.EncodingMode#TEMPLATE_AND_VALUES}.
+	 * @param encodingMode the mode to use for uri encoding
+	 * @since 5.3
+	 */
+	public void setEncodingMode(DefaultUriBuilderFactory.EncodingMode encodingMode) {
```

Comment:
Why not `setUriBuilderFactory` ?

Reply:
Right. I'm going to document that for more complex scenarios it would be better to configure an external `RestTemplate` and already use its `setUriTemplateHandler()`.

I would even go with all the `RestTempalte` options as an external reference. I'm not a fun of mutating internal objects.
But that would be too drastic even in the point release...

Reply:
Ok, makes sense; proceed with the remaining work 😄